dawn/renderer/utils/lcm_ws_bridge.js

```diff
@@ -0,0 +1,148 @@
+/*
+  LCM WebSocket Bridge: javascript client library
+
+  This library imitates the LCM APIs, and fulfills requests by forwarding them
+  to a websocket server.
+
+  PUBLIC API:
+
+    new LCM(websocket_url)
+    on_ready(callback)
+    subscribe(channel, msg_type, callback)
+    unsubscribe(subscription_id)
+    publish(channel, msg)
+
+  EXAMPLE CODE:
+
+    let l = new LCM("ws://localhost:8000");
+
+    l.on_ready(function() {
+      let sub = l.subscribe("sprocket/health", "health", function(msg) {
+        alert(msg.header.seq);
+      });
+
+      window.setTimeout(function() {
+        l.unsubscribe(sub);
+      }, 1000);
+
+      l.publish("/forest/cmd", {
+        __type__: "forest_cmd",
+        header: {
+          __type__: "header",
+          seq: 5,
+          time: 0
+        },
+        lights: [[true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false],
+                 [true, true, false]],
+        servos: [0, 0, 0, 0, 0, 0, 0, 0]
+      });
+    });
+*/
+import WebSocket from 'ws';
+
+
+const CircularJSON = require('circular-json');
```

Comment:
Quirk might have been when communicating with station, since I cannot reproduce it right now (i.e. never tested FC setup on Mac). On quick check now, it looks slight different and without circularity, but not sure.